src/ed25519.h

```diff
@@ -24,12 +24,36 @@ extern "C" {
 int ED25519_DECLSPEC ed25519_create_seed(unsigned char *seed);
 #endif
 
+#include "sha512.h"
+#include "ge.h"
+typedef struct
+{
+    sha512_context hash_ct;
+    unsigned char r[64];
+} signature_context;
+typedef struct
+{
+    sha512_context hash_ct;
+    ge_p3 A;
+    int bool_number;
+    unsigned char *sig;
+} verify_context;
+
 void ED25519_DECLSPEC ed25519_create_keypair(unsigned char *public_key, unsigned char *private_key, const unsigned char *seed);
 void ED25519_DECLSPEC ed25519_sign(unsigned char *signature, const unsigned char *message, size_t message_len, const unsigned char *public_key, const unsigned char *private_key);
 int ED25519_DECLSPEC ed25519_verify(const unsigned char *signature, const unsigned char *message, size_t message_len, const unsigned char *public_key);
 void ED25519_DECLSPEC ed25519_add_scalar(unsigned char *public_key, unsigned char *private_key, const unsigned char *scalar);
 void ED25519_DECLSPEC ed25519_key_exchange(unsigned char *shared_secret, const unsigned char *public_key, const unsigned char *private_key);
 
+void ED25519_DECLSPEC ed25519_create_public_from_private(unsigned char *public_key, unsigned char *private_key);
+void ED25519_DECLSPEC ed25519_sign_init1(signature_context *sig_status, const unsigned char *private_key);
+void ED25519_DECLSPEC ed25519_sign_update1(signature_context *sig_status, const unsigned char *message_block, size_t block_len);
+void ED25519_DECLSPEC ed25519_sign_final1_init2(signature_context *sig_status, unsigned char *signature, const unsigned char *public_key);
```

Comment:
The original API is clear enough that one can easily deduce what the functions do from their name, even without documentation. I have no idea what `ed25519_sign_final1_init2` would even mean. Either this API doesn't make sense, or it needs a lot more explaining.